apollo-router/src/services/subgraph_service.rs

```diff
@@ -297,10 +304,8 @@ impl tower::Service<SubgraphRequest> for SubgraphService {
 
                         // If not then put the subscription_id in the extensions for callback mode and continue
                         // Do this if the topic doesn't already exist
-                        let callback_url = public_url.join(&format!(
-                            "{}/{subscription_id}",
-                            path.as_deref().unwrap_or("/callback")
-                        ))?;
+                        let callback_url =
+                            public_url.join(&format!("/callback/{subscription_id}"))?;
```

Comment:
ok this looks like a merge fail.

Reply:
[ab60f8c](https://github.com/apollographql/router/pull/3328/commits/ab60f8c8017eba96c819c7f9c8e94f074c79892d)